.github/workflows/build.yml

```diff
@@ -0,0 +1,54 @@
+name: Build
+
+on: push
+
+jobs:
+  windows_build:
+    name: Windows Build
+    runs-on: windows-latest
+    env:
+      BUILD_TYPE: Release
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          ref: ${{ github.event.pull_request.head.sha }} # Check out the code of the PR head
```

Comment:
question (non-blocking): wondering what this `ref: ...` line is for? I get the following warning with the `Github Actions` extension for vscode: 

![image](https://github.com/Open-Agriculture/AgIsoDDOPGenerator/assets/29043784/a9f116db-9017-4410-94b8-8ca616965096)

I think due to the fact that a `event.pull_request` is not always available on a `push` only trigger. E.g. a push can occur on a branch for which a pull request is yet to be opened

Reply:
I think it was a copy-paste from the main stack's file. I think I can probably remove it?